build-system/tasks/build-bento.js

```diff
@@ -0,0 +1,225 @@
+    } else if (argv.extensions === 'inabox') {
+      argv.extensions = INABOX_EXTENSION_SET.join(',');
+    }
+    const explicitComponents = argv.extensions.replace(/\s/g, '').split(',');
+    componentsToBuild = dedupe(componentsToBuild.concat(explicitComponents));
+  }
+  if (argv.extensions_from) {
+    const componentsFrom = getExtensionsFromArg(argv.extensions_from);
+    componentsToBuild = dedupe(componentsToBuild.concat(componentsFrom));
+  }
+  if (
+    !preBuild &&
+    !argv.nocomponents &&
+    !argv.extensions &&
+    !argv.extensions_from &&
+    !argv.core_runtime_only
+  ) {
+    const allComponents = Object.values(COMPONENTS).map((c) => c.name);
+    componentsToBuild = dedupe(componentsToBuild.concat(allComponents));
+  }
+  return componentsToBuild;
+}
+
+/**
+ * Watches for non-JS changes within an extensions directory to trigger
+ * recompilation.
+ *
+ * @param {string} componentsDir
+ * @param {string} name
+ * @param {string} version
+ * @param {boolean} hasCss
+ * @param {?Object} options
+ * @return {Promise<void>}
+ */
+async function watchComponent(componentsDir, name, version, hasCss, options) {
+  /**
+   * Steps to run when a watched file is modified.
+   */
+  function watchFunc() {
+    buildComponent(name, version, hasCss, {
+      ...options,
+      continueOnError: true,
+      isRebuild: true,
+      watch: false,
+    });
+  }
+
+  const cssDeps = `${componentsDir}/**/*.css`;
+  const jisonDeps = `${componentsDir}/**/*.jison`;
+  const ignored = /dist/; //should not watch npm dist folders.
+  watch([cssDeps, jisonDeps], {ignored}).on(
+    'change',
+    debounce(watchFunc, watchDebounceDelay)
+  );
+}
+
+/**
+ * Copies components from
+ * src/bento/components/$name/$name.js
+ * to
+ * dist/v0/$name-$version.js
+ *
+ * Optionally copies the CSS at components/$name/$version/$name.css into
+ * a generated JS file that can be required from the components as
+ * `import {CSS} from '../../../build/$name-0.1.css';`
+ *
+ * @param {string} name Name of the extension. Must be the sub directory in
+ *     the components directory and the name of the JS and optional CSS file.
+ * @param {string} version Version of the extension. Must be identical to
+ *     the sub directory inside the extension directory
+ * @param {boolean} hasCss Whether there is a CSS file for this extension.
+ * @param {?Object} options
+ * @param {!Array=} extraGlobs
+ * @return {!Promise<void>}
+ */
+async function buildComponent(name, version, hasCss, options = {}, extraGlobs) {
+  options.extraGlobs = extraGlobs;
+  options.npm = true;
+  options.bento = true;
+
+  if (options.compileOnlyCss && !hasCss) {
+    return;
+  }
+  const componentsDir = `src/bento/components/${name}/${version}`;
+  if (options.watch) {
+    await watchComponent(componentsDir, name, version, hasCss, options);
+  }
+
+  const promises = [];
+  if (hasCss) {
+    if (!existsSync('build/css')) {
+      mkdirSync('build/css', {recursive: true});
+    }
```

Comment:
`mkdir`/`mkdirSync` don't need the `exists` check when `recursive` is used
```suggestion
    mkdirSync('build/css', {recursive: true});
```